packages/opencensus-exporter-instana/test/test-instana.ts

```diff
@@ -0,0 +1,95 @@
+/**
+ * Copyright 2018 OpenCensus Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {classes, types} from '@opencensus/opencensus-core';
+import * as assert from 'assert';
+import * as http from 'http';
+import * as mocha from 'mocha';
+import * as nock from 'nock';
+
+import {InstanaTraceExporter} from '../src/instana';
+
+/** Interface with request response model */
+interface RequestResponse {
+  statusCode: number;
+  statusMessage: string;
+}
+
+/**
+ * Controls if the tests will use a real network or not
+ * true to use a real Instana agent
+ * false to use a nock server
+ */
+const OPENCENSUS_NETWORK_TESTS =
+    ['true', 'TRUE', '1'].indexOf(process.env.OPENCENSUS_NETWORK_TESTS) > -1;
+
+/** Default config for traces tests */
+const defaultConfig: types.TracerConfig = {
+  samplingRate: 1
+};
+
+/** Run a nock server to replace the Instana agent */
+const runNockServer = () => {
+  nock('http://127.0.0.1:42699')
+      .persist()
+      .post('/com.instana.plugin.generic.trace')
+      .reply(200);
+};
+
+/** Checking if tests will use a real network, otherwise run a nock server */
+before(() => {
+  if (!OPENCENSUS_NETWORK_TESTS) {
+    runNockServer();
+  }
+});
+
+describe('Instana Exporter', function() {
+  /** Desabling the timeout for tests */
+  this.timeout(0);
+
+  describe('onEndSpan()', () => {
```

Comment:
I know this test is present in other unit tests, but I think it is not necessary. There is nothing unique about how the Instana exporter queues traces.

Reply:
Totally fine with that 👍